demos/benchmark/lv_demo_benchmark.c

```diff
@@ -548,34 +688,39 @@ static void load_scene(uint32_t scene)
     lv_anim_delete(lv_layer_top(), color_anim_cb);
     lv_obj_set_style_bg_opa(lv_layer_top(), LV_OPA_TRANSP, 0);
 
-    rnd_reset();
-    if(scenes[scene].create_cb) scenes[scene].create_cb();
+    rnd_reset(context);
+    if(scenes[context->scene_act].create_cb) {
+        scenes[context->scene_act].create_cb(context);
+    }
 }
 
 static void next_scene_timer_cb(lv_timer_t * timer)
 {
-    LV_UNUSED(timer);
+    benchmark_context_t * context = (benchmark_context_t *)timer->user_data;
 
-    scene_act++;
+    context->scene_act++;
 
-    load_scene(scene_act);
-    if(scenes[scene_act].scene_time == 0) {
+    load_scene(context);
+    if(scenes[context->scene_act].scene_time == 0) {
+        benchmark_context_deinit(context);
+        lv_free(context);
         lv_timer_delete(timer);
         summary_create();
     }
     else {
-        lv_timer_set_period(timer, scenes[scene_act].scene_time);
+        lv_timer_set_period(timer, scenes[context->scene_act].scene_time);
     }
 }
 
 #if LV_USE_PERF_MONITOR
 static void sysmon_perf_observer_cb(lv_observer_t * observer, lv_subject_t * subject)
 {
     const lv_sysmon_perf_info_t * info = lv_subject_get_pointer(subject);
-    char scene_name[64];
+    benchmark_context_t * context = (benchmark_context_t *)(observer->user_data);
```

Comment:
No cast is required for `void*` types, and please use functions to access object data.
```suggestion
    benchmark_context_t * context = lv_observer_get_user_data(observer);
```